src/lightning/fabric/strategies/fsdp.py

```diff
@@ -254,10 +255,13 @@ def module_to_device(self, module: Module) -> None:
         pass
 
     @contextmanager
-    def module_init_context(self) -> Generator[None, None, None]:
+    def module_init_context(self, empty_weights: Optional[bool] = None) -> Generator[None, None, None]:
         # TODO: Use the meta device and reset parameters after https://github.com/pytorch/pytorch/issues/90465
```

Comment:
Is this TODO still relevant after this addition?

Reply:
Yes, I would prefer to be a bit smarter about the initialization in case of FSDP in the future, either meta device or if the tensor materialization from torchdistx makes it into pytorch. Right now, FSDP has a bit of a buggy/unfinished situation when it comes to materialization, especially when user has custom parameter initialization function.

Reply:
The todo doesn't strictly need to be there, I can also remove it and make a GH issue instead.